Applications/VpView/vtkVpTrackModel.cxx

```diff
@@ -0,0 +1,81 @@
+/*ckwg +5
+ * Copyright 2017 by Kitware, Inc. All Rights Reserved. Please refer to
+ * KITWARE_LICENSE.TXT for licensing information, or contact General Counsel,
+ * Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.
+ */
+
+#include "vtkVpTrackModel.h"
+
+#include <vtkObjectFactory.h>
+
+vtkStandardNewMacro(vtkVpTrackModel);
+
+//-----------------------------------------------------------------------------
+vtkVpTrackModel::vtkVpTrackModel() : vtkVgTrackModel()
+{
+}
+
+//-----------------------------------------------------------------------------
+vtkVpTrackModel::~vtkVpTrackModel()
+{
+}
+
+//-----------------------------------------------------------------------------
+void vtkVpTrackModel::RemoveTrack(vtkIdType trackId)
+{
+  this->vtkVgTrackModel::RemoveTrack(trackId);
+  this->TrackKeyframes.erase(trackId);
+}
+
+//-----------------------------------------------------------------------------
+void vtkVpTrackModel::AddKeyframe(
+  vtkIdType trackId, const vtkVgTimeStamp& timeStamp)
+{
+  auto& trackKeyframes = this->TrackKeyframes[trackId];
+  trackKeyframes.insert(timeStamp);
+}
+
+//-----------------------------------------------------------------------------
+void vtkVpTrackModel::RemoveKeyframe(
+  vtkIdType trackId, const vtkVgTimeStamp& timeStamp)
+{
+    auto& trackKeyframes = this->TrackKeyframes[trackId];
+    trackKeyframes.erase(timeStamp);
+}
+
+//-----------------------------------------------------------------------------
+bool vtkVpTrackModel::GetIsKeyframe(
+  vtkIdType trackId, const vtkVgTimeStamp& timeStamp) const
+{
+  const auto& trackKeyframesIter = this->TrackKeyframes.find(trackId);
+  if (trackKeyframesIter != this->TrackKeyframes.end())
+    {
+    const auto& trackKeyframes = trackKeyframesIter->second;
+    const auto& keyframeIter = trackKeyframes.find(timeStamp);
+    return keyframeIter != trackKeyframes.end();
+    }
+  return false;
+}
+
+//-----------------------------------------------------------------------------
+void vtkVpTrackModel::SetTrackId(vtkVgTrack* track, vtkIdType newId)
+{
+  const auto oldId = track->GetId();
+
+  this->RemoveTrack(oldId);
+  track->SetId(newId);
+  this->AddTrack(track);
+
+  const auto& tkfIter = this->TrackKeyframes.find(oldId);
```

Comment:
Isn't his always going to return `this->TrackKeyframes.end()` since call to `RemoveTrack(oldId)` [here](https://github.com/Kitware/vivia/pull/10/files#diff-747ec5f01dae1537ff75758fd6e3263bR65) results it it being erased?

Reply:
Um... yes... :flushed: